### PR TITLE
Ignored files incorrect outside root

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
Duplicates #160 

Recreated to be on the proper branch.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/autoupgrade/165)

### CODE REVIEW

File: classes/TaskRunner/Upgrade/UpgradeFiles.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ protected function listFilesToUpgrade($dir)
         foreach ($allFiles as $file) {
             $fullPath = $dir . DIRECTORY_SEPARATOR . $file;
 
-            if ($this->container->getFilesystemAdapter()->isFileSkipped($file, $fullPath, 'upgrade')) {
+            if ($this->container->getFilesystemAdapter()->isFileSkipped($file, $fullPath, 'upgrade', $this->container->getProperty(UpgradeContainer::LATEST_PATH))) {
```
Comment (PierreRambaud) [inline classes/TaskRunner/Upgrade/UpgradeFiles.php]: I know you hate me, but line really long :/

File: classes/UpgradeTools/FilesystemAdapter.php [hunk 2]
```diff
@@ -202,56 +202,37 @@ public function listSampleFiles($dir, $fileext = '.jpg')
...
-                // note the restore process use skipFiles only if xml md5 files
-                // are unavailable
-            case 'restore':
-                if (in_array($file, $this->fileFilter->getExcludeFiles())) {
-                    return true;
-                }
+        if (null !== $temporaryWorkspace) {
+            $rootpath = str_replace('\\', '/', $temporaryWorkspace);
+        } else {
+            $rootpath = str_replace('\\', '/', $this->prodRootDir);
```
Comment (PierreRambaud) [inline classes/UpgradeTools/FilesystemAdapter.php]:
```php
$rootpath = str_replace('\\', '/', null !== $temporaryWorkspace ? $temporaryWorkspace : $this->prodRootDir);
```

Comment (Quetzacoalt91) [reply classes/UpgradeTools/FilesystemAdapter.php]: I chose to not write as you suggested. This makes the line quite difficult to read.

Comment (PierreRambaud) [reply classes/UpgradeTools/FilesystemAdapter.php]: And like this ? :D

```php
$rootpath = str_replace(
    '\\', 
    '/', 
    null !== $temporaryWorkspace ? $temporaryWorkspace : $this->prodRootDir
);
```

Comment (Quetzacoalt91) [reply classes/UpgradeTools/FilesystemAdapter.php]: Meh, ok!

**Pull request merged**